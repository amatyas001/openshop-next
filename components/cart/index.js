import { CartContent } from './CartContent/CartContent';
import { CartAmount } from './CartAmount/CartAmount';
import { CartButton } from './CartButton/CartButton';
import { CartItem } from './CartItem/CartItem';
import { CartPanel } from './CartPanel/CartPanel';

export { CartContent, CartAmount, CartButton, CartItem, CartPanel };
