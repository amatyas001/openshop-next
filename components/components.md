Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
