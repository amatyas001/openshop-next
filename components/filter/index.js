export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
