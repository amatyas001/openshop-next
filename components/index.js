export * from './shared';
export * from './cart';
export * from './payment';
