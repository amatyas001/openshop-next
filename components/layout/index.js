export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
