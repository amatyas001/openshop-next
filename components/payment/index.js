export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
