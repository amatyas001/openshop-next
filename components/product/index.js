import { ProductCard } from './ProductCard/ProductCard';
import { ProductControls } from './ProductControls/ProductControls';
import { ProductControlsColor } from './ProductControlsColor/ProductControlsColor';
import { ProductControlsAmount } from './ProductControlsAmount/ProductControlsAmount';
import { ProductControlsButton } from './ProductControlsButton/ProductControlsButton';
import { ProductControlsSize } from './ProductControlsSize/ProductControlsSize';

export {
  ProductCard,
  ProductControls,
  ProductControlsColor,
  ProductControlsAmount,
  ProductControlsButton,
  ProductControlsSize,
};
