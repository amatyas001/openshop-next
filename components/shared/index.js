import { Button } from './Button/Button';
import { ButtonIcon } from './ButtonIcon/ButtonIcon';
import { Field } from './Field/Field';
import { Image } from './Image/Image';
import { Panel } from './Panel/Panel';
import { Spinner } from './Spinner/Spinner';

export { Button, ButtonIcon, Field, Image, Panel, Spinner };
