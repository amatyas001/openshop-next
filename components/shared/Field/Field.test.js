import { create, act } from 'react-test-renderer';
import { Field } from '@app/components';

let tree;
const handler = jest.fn();
const types = ['text', 'email', 'password'];
const props = {
  id: 'mock_id',
  label: 'mock_label',
  onChange: handler,
  placeholder: 'mock_placeholder',
  type: 'text',
};

describe('<Field />', () => {
  beforeAll(() => {
    tree = create(<Field {...props} />);
  });

  it('should render with required props', () => {
    expect.assertions(1);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should call handler', async () => {
    expect.assertions(1);
    await act(async () => {
      tree.root.findByType('input').props.onChange('');
    });
    expect(handler).toHaveBeenCalledTimes(1);
  });

  it('should render proper label', () => {
    expect.assertions(3);
    expect(tree.root.findByType('label').props.htmlFor).toEqual(props.id);
    expect(tree.root.findByType('input').props.id).toEqual(props.id);
    expect(tree.root.findByType('label').children).toContain(props.label);
  });

  types.forEach((type) => {
    it(`should render ${type} input`, () => {
      expect.assertions(1);
      tree = create(<Field {...props} type={type} />);
      expect(tree.root.findByType('input').props.type).toEqual(type);
    });
  });
});
