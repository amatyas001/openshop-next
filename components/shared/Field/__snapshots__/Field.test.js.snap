// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> should render with required props 1`] = `
<fieldset
  className="css-9x3bhi"
>
  <label
    className="css-1cj26ek"
    htmlFor="mock_id"
  >
    mock_label
  </label>
  <input
    autoComplete={false}
    className="css-185v6vq"
    id="mock_id"
    onChange={[MockFunction]}
    placeholder="mock_placeholder"
    required={false}
    type="text"
    value={null}
  />
</fieldset>
`;
